src/github.com/matrix-org/rageshake/submit.go

```diff
@@ -41,10 +41,12 @@ type submitServer struct {
 }
 
 type payload struct {
-	Text      string     `json:"text"`
-	Version   string     `json:"version"`
-	UserAgent string     `json:"user_agent"`
-	Logs      []logEntry `json:"logs"`
+	Text      string            `json:"text"`
+	AppName   string            `json:"app"`
+	Version   string            `json:"version"`
+	UserAgent string            `json:"user_agent"`
+	Logs      []logEntry        `json:"logs"`
+	Data      map[string]string `json:"data"`
```

Comment:
hrm. I'm not keen on letting it be arrays/objects, since all we end up doing is figuring out how to turn it into a string. It would be nice if it stringified numbers automatically: any idea if there's a neat way to achieve this?